connector/src/main/java/org/jboss/as/connector/services/resourceadapters/DirectConnectionFactoryActivatorService.java

```diff
@@ -145,21 +145,17 @@ public void start(org.jboss.msc.service.StartContext context) throws org.jboss.m
             Map<String, String> mcfConfigProperties = new HashMap<String, String>();
             String securitySetting = null;
             String securitySettingDomain = null;
-            boolean elytronEnabled = false;
 
             if (properties != null) {
                 for (Map.Entry<String,String> prop : properties.entrySet()) {
                     String key = prop.getKey();
                     String value = prop.getValue();
                     if (key.equals("ironjacamar.security")) {
                         securitySetting = value;
-                    } else if (key.equals("ironjacamar.security.elytron") && value.equals("true")) {
-                        elytronEnabled = true;
                     } else if (key.equals("ironjacamar.security.elytron-authentication-context")) {
                         securitySettingDomain = value;
-                        elytronEnabled = true;
                     } else if (key.equals("ironjacamar.security.domain")) {
-                        securitySettingDomain = value;
+                        SUBSYSTEM_RA_LOGGER.legacySecurityNotSupported();
```

Comment:
This method produces an OperationFailedException but it is not thrown. I don't think throwing OFE from here is correct; some other exception type is needed.

Reply:
fixed